페이스북 access token을 long lived access token으로 변경하였습니다.

genie-youn opened the pull request:
현재 Facebook access token을 account controller의 signup method와 usersapi controller의 register method 두군데서 사용하고 있는데 세션에 따로 저장하지 않고 필요할 때 마다 js에서 받아와 사용하려고 했습니다.

그런데 js에서 access token을 가져오는게 아니라 인증코드? 를 받아오고 이를 통해 access token을 가져오는것 같아서 확인해보니 하나의 인증코드로는 하나의 access token만을 발급받을 수 있는것 같아서.. long lived token으로 변경해 세션에 저장해두는걸로 구현하였습니다.

#304 의 This authorization code has been used error가 js에서 access token을 두번 가져오려고 할때 발생하는 에러인것 같습니다.

genie-youn committed:
Change longlivedtoken

genie-youn committed:
Change longlivedtoken

kimtree committed:
Change PK column in ci_sessions

kimtree commented on application/controllers/Account.php:
```diff
@@ -32,10 +32,10 @@ public function signup() {
             ]);
 
-            $accessToken = $this->accountlib->get_facebook_access_token();
+            $longLivedAccessToken = $this->accountlib->get_facebook_access_token();
```
이 토큰을 $accessToken = new Facebook\Authentication\AccessToken('{example-access-token}') 로 만든다음에 isLongLived()가 맞는지, getExpiresAt()이 길게 나오는지 확인 부탁드립니다. 그리고 longLivedToken이라고 해도 언젠가는 expired될 수 있습니다. 그런 경우에 예외처리도 필요합니다.

genie-youn replied:
확인 완료했습니다~

kimtree commented on application/controllers/Account.php:
```diff
@@ -32,10 +32,10 @@ public function signup() {
-            if ($accessToken !== NULL) {
-                $fb->setDefaultAccessToken($accessToken);
+            if ($longLivedAccessToken !== NULL) {
+                $fb->setDefaultAccessToken($longLivedAccessToken);
```
longLivedToken이 사용이 종료되면 session에서 지워주는 것도 필요할 것 같습니다. 아니면 가입할 때 해당 토큰을 DB에 저장하고 추후 활용할 수 있을 것 같습니다.

genie-youn replied:
토큰이 로그인시마다 재발급되기 때문에 DB에 따로 저장하지 않고 세션내 로그인 정보들과 같은 생명주기를 갖는게 좋지 않을까요?

kimtree committed:
Merge pull request #334 from yamujine/feature/#333_ci_session_dup_entry

Change PK column in ci_sessions

kimtree committed:
Change Twig library to autoescape based on ‘html’ (boolean option was deprecated in 1.x twig)

kimtree committed:
Allow not_authenticate to check forced updated user session for validation is_auth

kimtree committed:
Add a hook for login check

kimtree committed:
Add custom libraries to use autocomplete

kimtree committed:
Add an image min library and optimize images

junho85 reviewed (approved):
👍

kimtree committed:
Add a phpdoc

kimtree committed:
Merge pull request #337 from yamujine/feature/#328_login_check_hook

Add a hook for login check

kimtree committed:
Remove ‘twig’ library from controllers (because it’s already defined in autoload.php)

kimtree committed:
Fix indents and wrong phpdocs

kimtree committed:
Remove landing relative features

kimtree committed:
Add a phpdoc and remove deprecated functions

mintplo committed:
Merge pull request #339 from yamujine/feature/#338_image_optimizer

Add an image min library and optimize images

kimtree committed:
Minify css

kimtree committed:
Make a verify route can be access from the person who is not authenticated

junho85 committed:
장소 - 내 작품 전시지원하기 - Filter "thumb_url" is already registered.

kimtree committed:
Add an API to CRUD exhibitions

kimtree committed:
Merge pull request #348 from yamujine/#342_add_exhibition

Add an API to CRUD exhibitions

kimtree committed:
Merge pull request #347 from yamujine/bug/#345_apply

장소 - 내 작품 전시지원하기 - Filter "thumb_url" is already registered.

kimtree committed:
Make a page for managing exhibitions

kimtree committed:
Add a monkey patching code for updating exhibition by place id

junho85 committed:
Merge pull request #349 from yamujine/feature/#342_add_exhibition_page

Make a page for managing exhibitions

junho85 committed:
전시목록 1단계
https://github.com/yamujine/artmatch/issues/350#issuecomment-302940195

junho85 committed:
전시기간이랑 작품수 사이의 바
지원하기 버튼 > 이미지
https://github.com/yamujine/artmatch/issues/350#issuecomment-303118829

kimtree committed:
Add a title column to exhibition

kimtree committed:
Modify an API and place detail function to insert and update exhibition title

kimtree committed:
Add a column to write a reason to apply

nablyy committed:
Add a view when there is no exhibition

kimtree committed:
Make a urlgenerator creates links protocol insensitive

nablyy committed:
Fix a bug

nablyy committed:
jquery3.1.1

nablyy committed:
bower

junho85 committed:
브랜치를 안따서 브랜치 땀. 지금 전시중이에요! 작업중.

junho85 committed:
지금전시중이에요 마크업

nablyy committed:
npm

nablyy committed:
return bower

kimtree committed:
Changes file permissions

kimtree committed:
Optimize png images

kimtree committed:
Apply css cleaner(minifier) for all css in assets folder

kimtree committed:
Lock npm packages (npm shrinkwrap)

nablyy committed:
setting

nablyy committed:
Merge branch 'develop' into feature/#350_exhibition_list_of_place_detail

nablyy committed:
setting

nablyy committed:
Merge branch 'feature/#350_exhibition_list_of_place_detail' of github.com:yamujine/artmatch into feature/#350_exhibition_list_of_place_detail

nablyy committed:
Merge pull request #353 from yamujine/feature/#350_exhibition_list_of_place_detail

Add a view when there is no exhibition

junho85 committed:
Merge remote-tracking branch 'origin/develop' into feature/#350_place_exhibition

# Conflicts:
#	static/dist/css/places/detail.min.css

junho85 committed:
Merge pull request #355 from yamujine/feature/#350_place_exhibition

Feature/#350 place exhibition

kimtree committed:
Merge pull request #352 from yamujine/feature/apply_reason

Add a column to write a reason to apply

kimtree committed:
Merge pull request #351 from yamujine/feature/add_exhibition_title

Feature/add exhibition title

nablyy committed:
Add script on addButton

nablyy committed:
bower

nablyy committed:
Add a button name

nablyy committed:
Change a button tag into an a tag

kimtree committed:
Update minified css and optimized css

nablyy committed:
bower

nablyy committed:
Edit commit files

nablyy committed:
Merge pull request #360 from yamujine/feature/#356_exhibition_list_in_place_detail

Feature/#356 exhibition list in place detail

kimtree committed:
Fix a bug where query can't show multiple exhibitions per place

kimtree committed:
Fix a bug that form_validation library not loaded

genie-youn committed:
Check token is available

genie-youn committed:
Merge branches 'enhancement/#311_remove_facebook_access_token_session' and 'enhancement/#311_remove_facebook_access_token_session' of https://github.com/yamujine/artmatch into enhancement/#311_remove_facebook_access_token_session

genie-youn commented:
향후 FB access token이 언제 또 사용될지 생각해보았습니다.
기획엔 없지만 아마 전시내용이나 작품을 페이스북 타임라인에 공유하기 정도로 사용될 것 같은데,

세션에서 토큰을 가져올때 라이브러리에서 만료된 토큰일 경우 null을 반환하고
이를 사용하는 비즈니스 로직에서 토큰이 없을경우 페이스북 로그인을 하게 만드는게 좋을 것 같아서 일단 null을 반환하는 정도로 수정하였습니다.

현재 세션 유지 시간은
CI 세션 유지 시간 (2시간) / 페이스북 LongLivedToken의 유지 시간 (2개월) 입니다.

kimtree commented:
diff보기가 좀 어려운데 머지 또는 rebase 좀 부탁드립니다~
